network/hnswrapper/hnsv2wrapperfake.go

```diff
@@ -7,21 +7,149 @@
 package hnswrapper
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
 	"github.com/Microsoft/hcsshim/hcn"
 )
 
+const networkName = "azure"
+
+var errorFakeHNS = errors.New("errorFakeHNS Error")
+
+func newErrorFakeHNS(errStr string) error {
+	return fmt.Errorf("%w : %s", errorFakeHNS, errStr)
+}
+
 type Hnsv2wrapperFake struct {
+	Cache FakeHNSCache
+	*sync.Mutex
```

Comment:
Lint was not happy that i was passing a value and not reference.

Reply:
Interesting.. good to know why.

Reply:
I haven't looked at the usage, but probably because you should never copy a `sync.Mutex`. In a fake this might be okay, but I would really discourage references to mutexes because any copy (like passed if you pass by val in to parallel goroutines) of the object can lock _all_ of them